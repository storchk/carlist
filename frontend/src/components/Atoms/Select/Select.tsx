import { Typography } from '../Typography'
import { StyledSelect } from './Select.styled'
import type { SelectProps } from './Select.types'

export const Select = ({ options, label, id, selected, ...props }: SelectProps) => {
  return (
    <StyledSelect>
      <label htmlFor={id}>
        <Typography
          fontSize={{
            mobile: 'xs',
          }}
          fontWeight="bold"
        >
          {label}
        </Typography>
      </label>
      <select id={id} defaultValue={selected} {...props}>
        {options.map((option, index) => (
          <option key={index} value={option.value?.toLocaleLowerCase()}>
            {option.label}
          </option>
        ))}
      </select>
    </StyledSelect>
  )
}
