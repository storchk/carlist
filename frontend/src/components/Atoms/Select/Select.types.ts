import type { ChangeEvent } from 'react'

export type SelectOptionProps = {
  value: string | null
  label: string
}
export type SelectProps = {
  label: string
  name: string
  id: string
  options: SelectOptionProps[]
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void
}
