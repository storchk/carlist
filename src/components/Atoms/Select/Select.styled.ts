import styled, { css } from 'styled-components'

export const StyledSelect = styled.div(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;

    select {
      margin-top: ${theme.spacing.xs};
      background-color: ${theme.colors.contrast300};
      padding: ${theme.spacing.sm};
      border-radius: ${theme.border.radius};

      &:hover {
        cursor: pointer;
      }
    }
  `
)
