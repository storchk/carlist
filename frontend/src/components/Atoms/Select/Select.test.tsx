import { axe } from 'jest-axe'

import { testing } from '../../../testing'
import { Select } from './Select'
import type { SelectProps } from './Select.types'

const { render, screen, renderer } = testing
const mockOptions: SelectProps['options'] = [
  {
    label: 'Lorem Ipsum',
    value: 'lorem-ipsum',
  },
]

const component = (
  <Select
    options={mockOptions}
    id="test-id"
    label="test-label"
    name="test-name"
    onChange={jest.fn}
  />
)

const renderComponent = () => render(component)
describe('atoms: Select', () => {
  it('should render Select with text', () => {
    renderComponent()
    const text = screen.getByText(/Lorem Ipsum/i)
    expect(text).toBeInTheDocument()
  })

  it('snapshot', () => {
    const view = renderer(component)
    expect(view).toMatchSnapshot()
  })

  it('a11y - No violations detected', async () => {
    const { container } = renderComponent()
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
