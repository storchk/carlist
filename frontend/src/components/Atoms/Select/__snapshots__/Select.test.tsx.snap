// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Select snapshot 1`] = `
.c1 {
  font-weight: 700;
  color: inherit;
  font-size: 14px;
}

.c0 {
  display: flex;
  flex-direction: column;
}

.c0 select {
  margin-top: 8px;
  background-color: #dee3ed;
  padding: 16px;
  border-radius: 8px;
}

.c0 select:hover {
  cursor: pointer;
}

<div
  className="c0"
>
  <label
    htmlFor="test-id"
  >
    <span
      className="c1"
      fontSize={
        {
          "mobile": "xs",
        }
      }
      fontWeight="bold"
    >
      test-label
    </span>
  </label>
  <select
    id="test-id"
    name="test-name"
    onChange={[Function]}
  >
    <option
      value="lorem-ipsum"
    >
      Lorem Ipsum
    </option>
  </select>
</div>
`;
